packages/headless/src/app/engine.ts

```diff
@@ -412,3 +412,16 @@ class InvalidEngineConfiguration extends Error {
     this.name = 'InvalidEngineConfiguration';
   }
 }
+
+export function warnIfUsingNextAnalyticsModeForServiceFeature(
```

Comment:
Oh well maybe here? You could export the text separately, to re-use it even in the tests?